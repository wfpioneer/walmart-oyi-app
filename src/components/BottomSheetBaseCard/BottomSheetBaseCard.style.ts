import { Dimensions, StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

const { width } = Dimensions.get('window');
const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    height: 75
  },
  touchableOpacity: {
    borderColor: COLOR.GREY_200,
    borderWidth: 1,
    width,
    flexDirection: 'row'
  },
  imageView: {
    height: 75,
    width: 75,
    alignItems: 'center',
    justifyContent: 'center'
  },
  image: {
    height: 30,
    width: 30
  },
  textView: {
    justifyContent: 'center'
  },
  text: {
    fontSize: 16
  }
});

export default styles;
