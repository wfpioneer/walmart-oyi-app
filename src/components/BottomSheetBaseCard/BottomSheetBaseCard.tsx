import { BottomSheetView, TouchableOpacity } from '@gorhom/bottom-sheet';
import React from 'react';
import { Image, Text } from 'react-native';
import styles from './BottomSheetBaseCard.style';

interface BaseCardProps {
  image: { uri: string },
  text: string,
  onPress: () => void
}

const BottomSheetBaseCard = (props: BaseCardProps): JSX.Element => {
  const { image, text, onPress } = props;

  return (
    <BottomSheetView style={styles.container}>
      <TouchableOpacity style={styles.touchableOpacity} onPress={onPress}>
        <BottomSheetView style={styles.imageView}>
          <Image
            style={styles.image}
            source={image}
            defaultSource={require('../../assets/images/placeholder.png')}
          />
        </BottomSheetView>
        <BottomSheetView style={styles.textView}>
          <Text style={styles.text}>{text}</Text>
        </BottomSheetView>
      </TouchableOpacity>
    </BottomSheetView>
  );
};

export default BottomSheetBaseCard;
