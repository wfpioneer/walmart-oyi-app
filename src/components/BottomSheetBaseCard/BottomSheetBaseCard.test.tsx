import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import BottomSheetBaseCard from './BottomSheetBaseCard';

describe('Base card render tests', () => {
  it('renders the bottom sheet base card with image', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <BottomSheetBaseCard
        image={require('../../assets/images/sams_logo.jpeg')}
        onPress={() => {}}
        text="hello"
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('renders the bottom sheet base card with material community icon', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <BottomSheetBaseCard
        onPress={() => {}}
        text="Heyo"
        materialIconName="pine-tree-fire"
      />
    );

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
