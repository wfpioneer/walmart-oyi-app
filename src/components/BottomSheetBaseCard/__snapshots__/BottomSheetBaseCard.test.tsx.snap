// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base card render tests renders the bottom sheet base card 1`] = `
<Memo(BottomSheetView)
  style={
    Object {
      "flexDirection": "row",
      "height": 75,
    }
  }
>
  <ForwardRef
    onPress={[Function]}
    style={
      Object {
        "borderColor": "#eeeeee",
        "borderWidth": 1,
        "flexDirection": "row",
        "width": 750,
      }
    }
  >
    <Memo(BottomSheetView)
      style={
        Object {
          "alignItems": "center",
          "height": 75,
          "justifyContent": "center",
          "width": 75,
        }
      }
    >
      <Image
        defaultSource=""
        source={
          Object {
            "testUri": "../../../src/assets/images/sams_logo.jpeg",
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
    </Memo(BottomSheetView)>
    <Memo(BottomSheetView)
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        hello
      </Text>
    </Memo(BottomSheetView)>
  </ForwardRef>
</Memo(BottomSheetView)>
`;
